test/misc.jl

```diff
@@ -967,12 +967,18 @@ end
 
 include("testenv.jl")
 
-
-let flags = Cmd(filter(a->!occursin("depwarn", a), collect(test_exeflags)))
-    local cmd = `$test_exename $flags --depwarn=yes deprecation_exec.jl`
-
-    if !success(pipeline(cmd; stdout=stdout, stderr=stderr))
-        error("Deprecation test failed, cmd : $cmd")
+@testset "deprecation_exec" begin
+    flags = Cmd(filter(a->!occursin("depwarn", a), collect(test_exeflags)))
+    cmd = `$test_exename $flags --depwarn=yes deprecation_exec.jl`
+    io = IOBuffer()
+    pipln = pipeline(cmd; stdout=io, stderr=io)
```

Comment:
I'm away from computer and can't check myself: can you have stdout and stderr writing to the same buffer? It's nice if that just works, I guess it'd be similar to `&>` redirection in shell.

Reply:
From a quick test, it seems to work fine.

```julia
julia> println(read("foo.jl", String))
println(stdout, "out1")
println(stderr, "err1")


println(stdout, "out2")
println(stderr, "err2")


println(stdout, "out3")
println(stderr, "err3")


julia> cmd = `julia foo.jl`;

julia> io = IOBuffer();

julia> run(pipeline(cmd; stdout=io, stderr=io));

julia> println(String(take!(io)))
out1
err1
out2
err2
out3
err3


julia>
```

Reply:
You are not allowed to use an IOBuffer as output, as such types are not stream-oriented, so they lack the required EOF capabilities. Use a BufferStream instead

Reply:
I've switched this PR to use `Base.BufferStream()`.

Reply:
@vtjnash Is it legal for me to send both stdout and stderr to the same `BufferStream`? E.g.

```
io = Base.BufferStream()

run(pipeline(cmd; stdout=io, stderr=io))

close(io)
```

Reply:
Not really. It will result in the program getting 2 different streams, while you probably wanted it to get only one file description

Reply:
Hmmm. What's the best way to capture the `stdout` and `stderr` from a command into a single string?